integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseCreateReferenceExec.scala

```diff
@@ -46,7 +47,7 @@ abstract class BaseCreateReferenceExec(
           NessieUtils.getCurrentRef(api, currentCatalog, catalog)
         } catch {
           case e: NessieNotFoundException =>
-            throw new NessieNotFoundException(
+            throw new NessieReferenceNotFoundException(
```

Comment:
Should we have `NessieReferenceNotFoundException` as a `case` type too (line 49)?

Reply:
Not sure actually. Quite some time ago. But yes: all the Spark things need a lot of love.